rust/lance-encoding/benches/decoder.rs

```diff
@@ -45,12 +45,15 @@ const PRIMITIVE_TYPES: &[DataType] = &[
 // schema doesn't yet parse them in the context of a fixed size list.
 const PRIMITIVE_TYPES_FOR_FSL: &[DataType] = &[DataType::Int8, DataType::Float32];
 
-const ENCODING_OPTIONS: EncodingOptions = EncodingOptions {
-    cache_bytes_per_column: 8 * 1024 * 1024,
-    max_page_bytes: 32 * 1024 * 1024,
-    keep_original_array: true,
-    buffer_alignment: 64,
-};
+fn default_encoding_options(version: LanceFileVersion) -> EncodingOptions {
+    EncodingOptions {
+        cache_bytes_per_column: 8 * 1024 * 1024,
+        max_page_bytes: 32 * 1024 * 1024,
+        keep_original_array: true,
+        buffer_alignment: 64,
+        support_large_chunk: version >= LanceFileVersion::V2_2,
```

Comment:
I like putting the setting here.  We have a lot of these checks just scattered throughout the code and I think it is cleaner here.  Maybe we could include the version in the options and do something like...

```
    EncodingOptions {
        ...,
        version
    }
}

impl EncodingOptions {
    pub fn support_large_chunk(&self) -> bool {
        self.version >= LanceFileVersion::V2_2
    }
}
```

This way we can add new version-dependent configs without adding new fields.  We could then add them as fields only if we have a need for users to disable them (even on a newer version).